python/ray/serve/_private/http_proxy.py

```diff
@@ -482,7 +483,8 @@ async def __call__(self, scope, receive, send):
                 scope["path"] = route_path.replace(route_prefix, "", 1)
                 scope["root_path"] = root_path + route_prefix
 
-            request_id = get_random_letters(10)
+            # RequestIdMiddleware makes sure scoep has x-request-id attribute
+            request_id = MutableHeaders(scope=scope).get("x-request-id")
             request_context_info = {
                 "route": route_path,
                 "request_id": request_id,
```

Comment:
Should we also change line 499 to respect this? (basically asking if "x-request-id" should take precedent)